habit-tracker/C11/asiu3209/project2/src/components/HabitForm.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useState } from "react";
+import { FormHabit } from "./HabitCard";
+function HabitForm({ addHabit }: { addHabit: (habit: FormHabit) => void }) {
+  //useState used for storing form values
+  const [formValue, setFormValue] = useState<FormHabit>({
+    //Initital starting state
+    name: "",
+    frequency: 1,
+    progress: 0,
+  });
+  function submitForm(e: React.FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    addHabit(formValue);
+    //resets form after submission
+    setFormValue({
+      name: "",
+      frequency: 1,
+      progress: 0,
+    });
+  }
+  //Onchange sets formValue as info is being inputed
+  return (
+    <form id="habit-form" onSubmit={submitForm}>
+      <label>Habit Name:</label>
+      <input
+        onChange={(e) => setFormValue({ ...formValue, name: e.target.value })}
+        type="text"
+        placeholder="Habit Name"
+        name="name"
+        value={formValue.name}
+        required
+      />
+      <label>Frequency:</label>
```

Comment:
Should have a `for` attribute referencing the name of the input the label describes. In React the prop is called `htmlFor`